File: 03_Exception/WriteChar.c
#include <stdio.h>

int main(void) {
    FILE *fp;
    char *text = "Hello, world.\n  from WriteChar.c";
    int i;

    fp = fopen("output.txt", "w");
    if (fp == NULL) {
        printf("ファイルを開けません。\n");
        return 1;
    }

    for (i = 0; text[i] != '\0'; i++) {
        fputc(text[i], fp);
    }

    fclose(fp);

    return 0;
}

File: 03_Exception/WriteChar.java
import java.io.*;

public class WriteChar {
    public static void main(String[] args) {
        try {
            FileWriter out = new FileWriter("output.txt");

            String lf = System.lineSeparator();
            String text = "Hello, World." + lf + "  from WriteChar.java";

            for (int i = 0; i < text.length(); i++) {
                out.write(text.charAt(i));
            }

            out.close();
        } catch (IOException e) {
            System.out.println("An error occurred.");
        }
    }
}

